### PR TITLE
Update TAN Submission Flow (EXPOSUREAPP-6595)

### PR DESCRIPTION
**Pull request by axelherbstreith**
This PR updates the Submission Flow when using a TAN to allow the new Deletion Warning Screen.

To test this PR:
- Validate that the submission flow for registering a test using a TAN is working correctly (with no test of same type)
- Validate that the Deletion Warning Screen is shown and the flow is correct when using a TAN (with test of same type already registered using TAN or QRCode) 
- Also validate when using a TAN, that after the test registration the `NoConsent` screen is shown and that the submission of the consent is working correctly (when using a TAN in general, consent to submit the keys is not granted before  registration, but afterwards) 

Please note: Comments will be addressed in a polishing PR

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/deletionwarning/SubmissionDeletionWarningFragment.kt [hunk 1]
```diff
@@ -85,14 +86,24 @@ class SubmissionDeletionWarningFragment : Fragment(R.layout.fragment_submission_
             binding.submissionQrCodeScanSpinner.isVisible = state.apiRequestState == ApiRequestState.STARTED
 
             if (ApiRequestState.SUCCESS == state.apiRequestState) {
-                if (state.testResult == CoronaTestResult.PCR_POSITIVE) {
-                    viewModel.triggerNavigationToSubmissionTestResultAvailableFragment()
-                } else {
-                    viewModel.triggerNavigationToSubmissionTestResultPendingFragment()
+
+                when (viewModel.getRegistrationType()) {
```
Comment (kolyaopahle) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/deletionwarning/SubmissionDeletionWarningFragment.kt:90]: Could you not implement all this logic in the viewmodel?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/tan/SubmissionTanViewModel.kt [hunk 2]
```diff
@@ -43,36 +46,52 @@ class SubmissionTanViewModel @AssistedInject constructor(
...
+            } else {
+                onTanSubmit(teletan)
+            }
+        }
+    }
+
+    private suspend fun onTanSubmit(teletan: Tan) {
+
+        try {
+            registrationState.postValue(ApiRequestState.STARTED)
```
Comment (kolyaopahle) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/tan/SubmissionTanViewModel.kt:75]: lets mark the whole `registrationState` and `registrationError` business for refactoring later

**Review by kolyaopahle — APPROVED** [review]
two small comments rest looks good to me

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/tan/SubmissionTanViewModel.kt (hunk 2 above)
Comment (axelherbstreith) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/tan/SubmissionTanViewModel.kt:75]: Yes I will :)

**Review by AlexanderAlferov — APPROVED** [review]
LGTM 🚀

**Pull request merged**